tests/metadata_stores/conftest.py

```diff
@@ -443,137 +447,43 @@ def test_something(config_with_truncation):
 
 
 @pytest.fixture(scope="session")
-def moto_s3_server():
-    """Start a moto S3 server for testing (session-scoped).
+def s3_endpoint_url() -> Generator[str, None, None]:
+    server = ThreadedMotoServer(ip_address="127.0.0.1", port=5000)
+    server.start()
+    yield "http://127.0.0.1:5000"
+    server.stop()
 
-    Runs moto's S3 server in a separate process to avoid GIL contention with
-    delta-rs (Rust) threads. This mirrors the behaviour of the moto CLI.
-    """
-    import os
-    import socket
-    import subprocess
-    import sys
-
-    import requests
-
-    try:
-        import moto  # noqa: F401
-    except ImportError:
-        pytest.skip("moto not installed")
-
-    # Find a free port (moto server doesn't support port=0)
-    def find_free_port():
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-            s.bind(("", 0))
-            s.listen(1)
-            port = s.getsockname()[1]
-        return port
-
-    port = find_free_port()
-    endpoint_url = f"http://127.0.0.1:{port}"
-
-    env = os.environ.copy()
-    env.setdefault("AWS_DEFAULT_REGION", "us-east-1")
-
-    process = subprocess.Popen(  # type: ignore[call-overload]
-        [
-            sys.executable,
-            "-m",
-            "moto.server",
-            "-H",
-            "127.0.0.1",
-            "-p",
-            str(port),
-        ],
-        stdout=subprocess.DEVNULL,
-        stderr=subprocess.DEVNULL,
-        env=env,
-    )
-
-    # Wait for server to be ready (3 second timeout)
-    for _ in range(30):
-        if process.poll() is not None:
-            process.terminate()
-            pytest.skip("Moto server process terminated while starting")
-        try:
-            response = requests.get(f"{endpoint_url}/moto-api/data.json", timeout=0.1)
-            if response.ok:
-                break
-        except requests.exceptions.RequestException:
-            time.sleep(0.1)
-    else:
-        process.terminate()
-        try:
-            process.wait(timeout=5)
-        except subprocess.TimeoutExpired:
-            process.kill()
-        pytest.skip("Moto server did not start in time")
 
-    yield {
-        "endpoint_url": endpoint_url,
-        "access_key": "testing",
-        "secret_key": "testing",
-        "region": "us-east-1",
-    }
-
-    process.terminate()
-    try:
-        process.wait(timeout=5)
-    except subprocess.TimeoutExpired:
-        process.kill()
-
-
-@pytest.fixture
-def moto_s3_bucket(moto_s3_server):
-    """Create a test S3 bucket (function-scoped).
-
-    Creates a unique bucket for each test. Moto server state persists
-    across tests (session-scoped), so cleanup is required.
+@pytest.fixture(scope="function")
+def s3_bucket(s3_endpoint_url: str) -> Generator[dict[str, Any], None, None]:
```

Comment:
let's make this `tuple[str, dict[str, Any]` and rename to `s3_bucket_and_so` 
also no need to return the boto3 resource